### PR TITLE
TODO cleared: Refactor create and update channel methods

### PR DESCRIPTION
**Pull request by kmehant**
Signed-off-by: K mehant <411843@student.nitandhra.ac.in>

### CODE REVIEW

**Comment by kmehant** [comment]
@dexhunter Please review!

**Comment by dexhunter** [comment]
@kmehant  Could you have a look at the [error log](https://dev.azure.com/Hyperledger/Fabric/_build/results?buildId=16725&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=5caf77c8-9b10-50ef-b5c7-ca89c63e1c86&l=1271), thanks!

File: hfc/fabric/client.py [hunk 1]
```diff
@@ -462,12 +461,16 @@ async def channel_create(self, orderer, channel_name, requestor,
...
-        if self.get_channel(channel_name):
-            msg = f"channel {channel_name} already existed when creating"
-            _logger.warning(msg)
-            raise Exception(msg)
+        if signatures is None:
+            signatures = []
+
+        channel = self.get_channel(channel_name)
+        if not channel:
+            _logger.warning(f"channel {channel_name} does not exist")
```
Comment (dexhunter) [inline hfc/fabric/client.py:472]: should return `error` there

**Comment by dexhunter** [comment]
@kmehant you can try locally by `make check` to see what's caused the problem.

File: hfc/fabric/client.py (hunk 1 above)
Comment (kmehant) [reply hfc/fabric/client.py:472]: Will fix it by this weekend 😅

File: hfc/fabric/client.py [hunk 2]
```diff
@@ -444,13 +444,12 @@ def get_channel(self, name):
...
-    async def channel_create(self, orderer, channel_name, requestor,
-                             config_yaml=None, channel_profile=None,
-                             config_tx=None):
+    # TODO pass envelope directly
+    def get_headers_create_or_update_channel(self, orderer, channel_name, requestor,
+                                             config_yaml=None, channel_profile=None,
+                                             config_tx=None, signatures=None):
         """
-        Create a channel, send request to orderer, and check the response
+        Get headers to send request to orderer, and check the response
```
Comment (yeasy) [inline hfc/fabric/client.py:452]: This method will prepare the headers that are used for creating or updating channel.
Should make clear in the doc string.

**Pull request closed without merging**